text/0000-rust-crate-ownership.md

```diff
@@ -0,0 +1,158 @@
+
+### Internal use
+“Internal use” crates should contain the following text near the top of the readme/documentation:
+
+> This crate is maintained by \[team\], primarily for use by \[rust project(s)\] and not intended for external use (except as a transitive dependency). This crate may make major changes to its APIs or be deprecated without warning.
+
+
+The "except as a transitive dependency" text should be included if the crate is a dependency of an intentional-artifact library ("transitively intentional").
+
+Security issues in transitively intentional libraries should be handled as if they were intentional artifacts.
+
+
+### Experiment
+
+“Experiment” crates should mention they are experiments. Experiment crates may be intended to be used in a scoped sort of way; so if they are intended to be used they should be clear about what they are guaranteeing.
+
+An example of such messaging would be text like:
+
+> This crate is maintained by \[team\] as a part of an experiment around \[thingy\]. We encourage people to try to use this crate in their projects and provide feedback through \[method\], but do not guarantee long term maintenance.
+
+or, for experiments that are not intended to be used at all:
+
+> This crate is maintained by \[team\] and is an internal experiment. We do not guarantee stability or long term maintenance, please use at your own risk.
+
+Ideally, experimental crates that are published for feedback purposes will have a document to link to that lists out the purpose, rough duration, and processes of the experiment.
+
+### Deprecated
+“Deprecated” crates should contain the following text near the top of the readme/documentation:
+
+> This crate is deprecated and not intended to be used.
+
+### Placeholder
+
+“Placeholder” crates should contain the following text in their published readme/documentation:
+
+> This crate is a functionally empty crate that exists to reserve the crate name of \[tool\]. It should not be used. 
+
+In general it is better to have an empty placeholder crate published instead of reserving the crate via yanking, so that there is a readme that helps
+
+
+### Expatriated
+
+It's unclear if any action should be taken on these beyond removing any semblance of officialness (including rust-lang/foo team owners). We currently have only one such crate (`rand`).
+
+These should by and large not be considered to be "team managed" crates; this category is in this RFC for completeness to be able to talk about expatriation as an end state.
+
+## Transitions and new crates
+
+Teams should feel free to create new crates in any of these categories; however "Intentional Artifact" crates should ideally be accompanied with an RFC. As we move towards having team charters, this will also involve a charter change. Teams should notify core@rust-lang.org when they've created such crates so that the core team may track these crates and ensure this policy is applied.
```

Comment:
> should ideally be accompanied with an RFC.

is this optional? If so can you give example scenarios to show when it would be okay to not create an RFC?

Reply:
I actually don't think it _should_ be optional, but I was also wary of imposing this on teams? But I do think that an RFC isn't a super high bar if the expectation is that it'll be a small thing.

Also this ties in to the ongoing charter work, _really_ what's being requested here is a change to a charter, which likely will require an FCP, if not an RFC. But the charter stuff is not defined well enough for this RFC to refer to it yet.

Reply:
I'd lean towards just changing it now to say that it requires an RFC and changing it to the charter update process once that is available. I see what you're saying about RFC's being an imposition but how many RFCs like this can we realistically expect between now and when the charter update process is well defined? My money's on 0.

Reply:
Seems fair. I'm going to leave a hook such that it calls out this potential future update (so updating it isn't a big deal).